src/Passes/docs/src/DeveloperGuide/ArchitectureOverview.md

```diff
@@ -0,0 +1,55 @@
+# Architecture Overview
+
+This document assumes familiarity with LLVM, LLVM intermediate representation (IR), LLVM passes and the `opt` which is used to apply passes to IRs. We also assume that the reader is familiar with the QIR specification, the conventions used for functions and its opaque types.We further assume that the reader is familiar with basic C++. This is in particular true later on when we dig into the details of building pass components. Throughout this document we will assume that any reference to a QIR, is a valid within the specification requirements.
```

Comment:
1. I would suggest making key fragments to refer to the corresponding web pages. Some of them I made below, but the others would still be nice to have (LLVM passes, `opt`).
```suggestion
This document assumes familiarity with LLVM, [LLVM intermediate representation (IR)](https://llvm.org/docs/LangRef.html), LLVM passes and the `opt` which is used to apply passes to IRs. We also assume that the reader is familiar with the [QIR specification](https://github.com/microsoft/qsharp-language/tree/main/Specifications/QIR), the conventions used for functions and its opaque types. We further assume that the reader is familiar with basic C++. This is in particular true later on when we dig into the details of building pass components. Throughout this document we will assume that any reference to a QIR, is a valid within the specification requirements.
```
2. Something is missing after "valid" in the fragment:
> is a valid within the specification requirements